### PR TITLE
smaller k optimization part-1

### PR DESCRIPTION
**Pull request by ramjana**
first part of smaller K optimization code
code change to determine number of global fetches in each unroll iteration
make ds_writes as last instruction and lgkmcnt changes for ds_write

### CODE REVIEW

File: Tensile/KernelWriter.py [hunk 1]
```diff
@@ -96,45 +98,85 @@ def makeSchedule(self, kernel, tensorParametersA, tensorParametersB, localWriteE
       readCnt = self.globalReadACode.middle.countType(Code.GlobalReadInst) + \
                 self.globalReadBCode.middle.countType(Code.GlobalReadInst)
       # reads and incs are scheduled in iters range(0...endIter)
-      endIter = readCnt + 2 # 2 for incA and incB
+      # treat incA and incB as independent tasks that can be in the same iter code as Load(s) 
+      # 
+      numTaskstoSched += readCnt;
+      numTaskstoSched += 2 # 2 for incA and incB
```
Comment (bensander) [inline Tensile/KernelWriter.py]: there are enable[] switches that can disable the increments (for profiling).  Should account for those here?

File: Tensile/KernelWriter.py [hunk 2]
```diff
@@ -767,20 +815,20 @@ def kernelBody( self, kernel, tensorParametersA, tensorParametersB ):
 
           waitGlobalRead = -1
           if kernel["PrefetchGlobalRead"] and isResetLroIter:
-            waitLocalWrite = 1
+            waitLocalWrite = 1
```
Comment (bensander) [inline Tensile/KernelWriter.py]: please remove trailing spaces

File: Tensile/KernelWriter.py [hunk 3]
```diff
@@ -767,20 +815,20 @@ def kernelBody( self, kernel, tensorParametersA, tensorParametersB ):
...
-            waitLocalWrite = 1
+            waitLocalWrite = 1 
           else:
             waitLocalWrite = -1
           waitLocalRead  = 1 if isResetLroIter else 0
 
         else: # not isResetLroIter
           waitGlobalRead = 1 if u==0 and kernel["PrefetchGlobalRead"] and kernel["PrefetchLocalRead"] else -1
-          waitLocalWrite = -1
+          waitLocalWrite = -1
```
Comment (bensander) [inline Tensile/KernelWriter.py]: ts

File: Tensile/KernelWriter.py [hunk 4]
```diff
@@ -716,12 +759,17 @@ def kernelBody( self, kernel, tensorParametersA, tensorParametersB ):
...
+        localReadsB = Code.Module()
         for iui in range(0,kernel["InnerUnroll"]):
           if self.enable["LocalRead"]:
             localReads.addText(self.comment("local read a"))
             localReads.addCode(self.localReadDo(kernel, plrIdx, iui, tensorParametersA))
             localReads.addText(self.comment("local read b"))
             localReads.addCode(self.localReadDo(kernel, plrIdx, iui, tensorParametersB))
+            #container for holding local read A & B elements for later re-ordering
+            localReadsA.addCode(self.localReadDo(kernel, plrIdx, iui, tensorParametersA))
+            localReadsB.addCode(self.localReadDo(kernel, plrIdx, iui, tensorParametersB))
```
Comment (bensander) [inline Tensile/KernelWriter.py]: looks good

**Comment by ramjana** [comment]
fixed all review comments

**Pull request closed without merging**